adapters/ads_adapter.py

```diff
@@ -1,22 +1,173 @@
-""" Beckhoff ADS adapter.
+"""Beckhoff ADS adapter.
 
-Downloads all beckhof ADS data and sends it to the GUI tool.
+Subscribes to Beckhof ADS variables and sends it to the GUI tool.
+
+Auto subscribe only works for DINT and LREAL for now.
+
+Example usage:
+    python ads_adapter.py --regex GVl.*temperature
 
 """
+from lognplot.client import LognplotTcpClient
+import pyads
+import time
+import ctypes
+import struct
+import datetime
+import re
+import argparse
+from collections import namedtuple
+from typing import Type
 
-# TODO: implement.
 
+class AdsClient:
+    Entry = namedtuple(
```

Comment:
Maybe we could give this thing a more descriptive name, such as `AdsVariableDescriptionEntry` or something along those lines.